File: Numbers with this digit inside/Numbers with this digit inside.js
function numbersWithDigitInside(x, d) {
  let result = [...Array(x).keys()].map(i => i+1).filter(v => String(v).includes(String(d)))
  return [result.length,result.reduce((a,b) => a+b,0),result.length===0 ? 0 : result.reduce((a,b) => a*b,1)]
}

File: Numbers with this digit inside/Numbers with this digit inside.swift
func numbersWithDigitInside(_ x: Int64, _ d: Int64) -> [Int64] {
    let result = (1...x).filter { String($0).contains(String(d)) }
    let mul: Int64 = result.isEmpty ? 0 : result.reduce(1 as Int64, *)
    return [Int64(result.count), result.reduce(0 as Int64, +), mul]
}
